File: backend/events/getEvent.mjs
import { GetCommand } from "@aws-sdk/lib-dynamodb";
import { db, EVENTS_TABLE } from "./shared/db.mjs";
import { ok, badRequest, notFound, serverError, cors } from "./shared/response.mjs";

export async function handler(event) {
  if (event.httpMethod === "OPTIONS") return cors();

  const { eventId } = event.pathParameters ?? {};
  if (!eventId) return badRequest("eventId is required");

  try {
    const result = await db.send(new GetCommand({ TableName: EVENTS_TABLE, Key: { eventId } }));
    if (!result.Item) return notFound("Event not found");
    return ok(result.Item);
  } catch (e) {
    console.error(e);
    return serverError();
  }
}

File: backend/events/getEvent.py
import logging

from shared.db import events_table
from shared.response import bad_request, cors, not_found, ok, server_error, with_cors

log = logging.getLogger()


@with_cors
def handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return cors()

    event_id = (event.get("pathParameters") or {}).get("eventId")
    if not event_id:
        return bad_request("eventId is required")

    try:
        result = events_table().get_item(Key={"eventId": event_id})
        item = result.get("Item")
        if not item:
            return not_found("Event not found")
        return ok(item)
    except Exception:
        log.exception("getEvent failed")
        return server_error()
